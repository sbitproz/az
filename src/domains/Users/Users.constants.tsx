import { ColumnsType } from "antd/es/table";
import { Dob, Location, User } from "../../api/hooks/users.types";
import { Avatar, Flex } from "antd";
import { spacings } from "@/styles/spacings";
import {
  CalendarOutlined,
  EnvironmentOutlined,
  PhoneOutlined,
} from "@ant-design/icons";
import { dateFormat } from "@/common/utils/dateFormat.utils";
import { IconWrapper, RowWrapper } from "./Users.styles";
import { LightText, BoldText } from "@/common/components/Typography/Typography";

export const pageSizeOptions = ["10", "25", "50"];

export const userColumns: ColumnsType<User> = [
  {
    title: "User",
    dataIndex: "userIdenifier",
    key: "userIdenifier",
    ellipsis: true,
    width: 200,
    render: (_, record: User) => {
      const {
        name: { title, first, last },
        picture,
        email,
      } = record;
      return (
        <Flex>
          <IconWrapper>
            <Avatar src={picture.thumbnail} />
          </IconWrapper>
          <div>
            <BoldText>
              {title} {first} {last}
            </BoldText>
            <LightText>{email}</LightText>
          </div>
        </Flex>
      );
    },
  },
  {
    title: "Location",
    dataIndex: "location",
    key: "location",
    ellipsis: true,
    width: 100,
    render: (location: Location) => {
      const { city, state } = location;
      return (
        <RowWrapper>
          <IconWrapper>
            <EnvironmentOutlined />
          </IconWrapper>
          <div>
            {city}, <LightText>{state}</LightText>
          </div>
        </RowWrapper>
      );
    },
  },
  {
    title: "DOB",
    dataIndex: "dob",
    key: "dob",
    ellipsis: true,
    width: 100,
    render: (dob: Dob) => {
      const { age, date } = dob;
      return (
        <RowWrapper>
          <IconWrapper>
            <CalendarOutlined />
          </IconWrapper>
          <div>
            <span style={{ marginRight: spacings.S4 }}>{dateFormat(date)}</span>
            <LightText>(Age {age})</LightText>
          </div>
        </RowWrapper>
      );
    },
  },
  {
    title: "Telephone",
    dataIndex: "telephone",
    key: "telephone",
    width: 100,
    render: (_, record) => {
      const { cell, phone } = record;
      return (
        <RowWrapper
          style={{
            alignItems: "center",
          }}
        >
          <IconWrapper>
            <PhoneOutlined />
          </IconWrapper>
          <div>
            <div>
              {cell} <LightText>(m)</LightText>{" "}
            </div>
            <div>
              {phone} <LightText>(p)</LightText>
            </div>
          </div>
        </RowWrapper>
      );
    },
  },
  {
    title: "Registered",
    dataIndex: "registered",
    key: "registered",
    ellipsis: true,
    width: 100,
    render: (dob: Dob) => {
      const { age, date } = dob;
      return (
        <RowWrapper>
          <IconWrapper>
            <CalendarOutlined />
          </IconWrapper>
          <div>
            <span style={{ marginRight: spacings.S4 }}>{dateFormat(date)}</span>
            <LightText>({age} years ago)</LightText>
          </div>
        </RowWrapper>
      );
    },
  },
];
