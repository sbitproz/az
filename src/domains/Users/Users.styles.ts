import { spacing } from "@/styles/spacings";
import styled from "@emotion/styled";
import { Header as HeaderBase } from "antd/es/layout/layout";
import { Flex } from "antd";
import { colors } from "@/styles/colors";

export const RowWrapper = styled(Flex)`
  flex-direction: row;
`;

export const IconWrapper = styled(Flex)`
  margin-right: ${spacing.S6}px;
`;

export const Header = styled(HeaderBase)`
height: 80px;
padding-top: 20px;
  margin-bottom: 50px;
  background: ${colors.white100};
`;
