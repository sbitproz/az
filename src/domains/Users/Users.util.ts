import { User } from "@/api/hooks/users.types";

export const getRowKey = (user: User) => {
  // Fake error test
  // throw new Error('Error');
  return `${user.id.name}-${user.id.value}`;
};
