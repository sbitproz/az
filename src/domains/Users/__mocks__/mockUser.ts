import { User } from "@/api/hooks/users.types";

export const mockUser = {
  name: {
    title: "Mr",
    first: "Sebastian",
    last: "Silva",
  },
  location: {
    city: "Boise",
    state: "New Jersey",
  },
  email: "sebastian.silva@example.com",
  dob: {
    date: "1956-05-21T04:02:40.864Z",
    age: 67,
  },
  registered: {
    date: "2021-06-15T14:13:29.119Z",
    age: 2,
  },
  phone: "(329) 449-6469",
  cell: "(848) 863-4315",
  id: {
    name: "SSN",
    value: "780-89-7852",
  },
  picture: {
    large: "https://randomuser.me/api/portraits/men/80.jpg",
    medium: "https://randomuser.me/api/portraits/med/men/80.jpg",
    thumbnail: "https://randomuser.me/api/portraits/thumb/men/80.jpg",
  },
} as User;
