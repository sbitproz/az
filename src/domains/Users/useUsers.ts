import { useUserApi } from "@/api/hooks/useUserApi";
import useOnMount from "@/common/hooks/useOnMount/useOnMount";
import { globalEnv } from "@/globalEnv";
import { TablePaginationConfig } from "antd";
import { useCallback, useEffect, useState } from "react";

export const useUsers = () => {
  const { isLoading, fetchUsers, data: users } = useUserApi();

  const [queryValues, setQueryValues] = useState({current: 1, pageSize: 15, search: ''});

  

  useOnMount(() => {
    const fetchData = async () => {
      await fetchUsers(1, 15, '');
    };

    fetchData();
  });

  // useEffect(() => {
  //   const fetchData = async () => {
  //     await fetchUsers(queryValues.current, queryValues.pageSize);
  //   };

  //   fetchData();
  // }, [
  //   queryValues
  // ])

  const onTableChange = useCallback(
    async (pagination: TablePaginationConfig) => {
      if (pagination.current && pagination.pageSize) {
        fetchUsers(pagination.current, pagination.pageSize, '');
      }
    },
    []
  );

  const onSearchQuery = useCallback(
    async (searchValue: string) => {

      if (searchValue) {
        fetchUsers(queryValues.current, queryValues.pageSize, searchValue);
      }
    },
    []
  );


  return {
    users,
    isLoading,
    onTableChange,
    onSearchQuery,
    totalUserCount: globalEnv.TOTAL_RECORD_COUNT,

  };
};
