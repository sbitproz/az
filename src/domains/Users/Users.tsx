import { Table } from "antd";
import { useUsers } from "./useUsers";
import { User } from "../../api/hooks/users.types";
import { pageSizeOptions, userColumns } from "./Users.constants";
import { getRowKey } from "./Users.util";
import { tableStyles } from "@/common/styles/table.styles";
import { UserSearch } from "./components/UserSearch/UserSearch";
import { Content, Header } from "./Users.styles";
import { ErrorBoundary } from "@components/ErrorBoundary/ErrorBoundary";
import { somethingWentWrong } from "@components/ErrorBoundary/ErrorBoundary.utils";
import { Title } from "@/common/components/Typography/Typography";

const userErrorLabel = somethingWentWrong("the user table");

export const Users = () => {
  const {
    onTableChange,
    onSearchQuery,
    users,
    isLoading,
    totalUserCount,
    queryValues,
  } = useUsers();

  return (
    <>
      <Header>
        <Title>Users</Title>
        <UserSearch onSearchQuery={onSearchQuery} />
      </Header>
      <Content>
        <ErrorBoundary label={userErrorLabel}>
          <Table<User>
            css={tableStyles}
            columns={userColumns}
            dataSource={users}
            rowKey={getRowKey}
            onChange={onTableChange}
            loading={isLoading}
            pagination={{
              total: totalUserCount,
              pageSizeOptions,
              pageSize: queryValues.pageSize,
              current: queryValues.current,
            }}
          />
        </ErrorBoundary>
      </Content>
    </>
  );
};
