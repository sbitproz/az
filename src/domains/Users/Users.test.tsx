import { render, screen } from "@testing-library/react";
import { Users } from "./Users";
import { useUsers } from "./useUsers";
import { getRowKey } from "./Users.util";
import { mockUser } from "./__mocks__/mockUser";
vi.mock("./components/UserSearch/UserSearch", () => ({
  UserSearch: () => <>User Search</>,
}));
vi.mock("./useUsers");
vi.mock("./Users.util");

type UseUsersProps = ReturnType<typeof useUsers>;

const mockOnSearchQuery = vi.fn();
const mockOnTableChange = vi.fn();

const mockUseUsersValues: UseUsersProps = {
  isLoading: false,
  onSearchQuery: mockOnSearchQuery,
  onTableChange: mockOnTableChange,
  queryValues: { current: 1, pageSize: 10, search: "" },
  totalUserCount: 1000,
  users: [mockUser],
};

vi.mocked(useUsers).mockReturnValue(mockUseUsersValues);

describe("Users", () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });

  it("should match snapshot", () => {
    const { container } = render(<Users />);

    expect(container).toMatchSnapshot();
  });

  it("should render user's name", () => {
    render(<Users />);

    expect(screen.getByText(/Sebastian/)).toBeInTheDocument();
    expect(screen.getByText(/Silva/)).toBeInTheDocument();
    expect(screen.getByText(/Mr/)).toBeInTheDocument();
  });

  it("should render user's dob  date and age", () => {
    render(<Users />);

    expect(screen.getByText("05/21/1956")).toBeInTheDocument();
    expect(screen.getByText("(Age 67)")).toBeInTheDocument();
  });

  it("should render user's location", () => {
    render(<Users />);

    expect(screen.getByText("Boise,")).toBeInTheDocument();
    expect(screen.getByText("New Jersey")).toBeInTheDocument();
  });

  it("should render user's telephone and cell numbers", () => {
    render(<Users />);

    expect(screen.getByText("(848) 863-4315")).toBeInTheDocument();
    expect(screen.getByText("(329) 449-6469")).toBeInTheDocument();
  });

  it("should render user's registered date and age", () => {
    render(<Users />);

    expect(screen.getByText("06/15/2021")).toBeInTheDocument();
    expect(screen.getByText("(2 years ago)")).toBeInTheDocument();
  });

  it("should render loading when is loading is true", () => {
    vi.mocked(useUsers).mockReturnValue({
      ...mockUseUsersValues,
      isLoading: true,
    });

    render(<Users />);

    const antTableSpinner = document.querySelector(".ant-spin-spinning");

    expect(antTableSpinner).toBeDefined();
  });

  it("should not render loading when is loading is false", () => {
    vi.mocked(useUsers).mockReturnValue({
      ...mockUseUsersValues,
      isLoading: false,
    });

    render(<Users />);

    const antTableSpinner = document.querySelector(".ant-spin-spinning");

    expect(antTableSpinner).toBeNull();
  });

  it("should show error when error occurs", () => {
    vi.mocked(useUsers).mockReturnValue({
      ...mockUseUsersValues,
      isLoading: false,
    });

    vi.mocked(getRowKey).mockImplementation(() => {
      throw new Error("Error");
    });

    render(<Users />);

    expect(screen.getByTestId("error-image")).toBeInTheDocument();
  });

  it("should render UserSearch component", () => {
    render(<Users />);

    expect(screen.getByText("User Search")).toBeInTheDocument();
  });
});
