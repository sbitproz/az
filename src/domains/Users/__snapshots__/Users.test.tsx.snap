// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Users > should match snapshot 1`] = `
<div>
  <header
    class="ant-layout-header css-1cfoh7z css-dev-only-do-not-override-1okl62o"
  >
    <h1
      class="ant-typography css-1id17ut css-dev-only-do-not-override-1okl62o"
    >
      Users
    </h1>
    User Search
  </header>
  <main
    class="ant-layout-content css-2xdp0l css-dev-only-do-not-override-1okl62o"
  >
    <div
      class="ant-table-wrapper css-3qyxdu-Users css-dev-only-do-not-override-1okl62o"
    >
      <div
        class="ant-spin-nested-loading css-dev-only-do-not-override-1okl62o"
      >
        <div
          class="ant-spin-container"
        >
          <div
            class="ant-table css-dev-only-do-not-override-1okl62o"
          >
            <div
              class="ant-table-container"
            >
              <div
                class="ant-table-content"
              >
                <table
                  style="table-layout: fixed;"
                >
                  <colgroup>
                    <col
                      style="width: 200px;"
                    />
                    <col
                      style="width: 100px;"
                    />
                    <col
                      style="width: 100px;"
                    />
                    <col
                      style="width: 100px;"
                    />
                    <col
                      style="width: 100px;"
                    />
                  </colgroup>
                  <thead
                    class="ant-table-thead"
                  >
                    <tr>
                      <th
                        class="ant-table-cell ant-table-cell-ellipsis"
                        scope="col"
                        title="User"
                      >
                        User
                      </th>
                      <th
                        class="ant-table-cell ant-table-cell-ellipsis"
                        scope="col"
                        title="Location"
                      >
                        Location
                      </th>
                      <th
                        class="ant-table-cell ant-table-cell-ellipsis"
                        scope="col"
                        title="DOB"
                      >
                        DOB
                      </th>
                      <th
                        class="ant-table-cell"
                        scope="col"
                      >
                        Telephone
                      </th>
                      <th
                        class="ant-table-cell ant-table-cell-ellipsis"
                        scope="col"
                        title="Registered"
                      >
                        Registered
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    class="ant-table-tbody"
                  >
                    <tr
                      class="ant-table-row ant-table-row-level-0"
                    >
                      <td
                        class="ant-table-cell ant-table-cell-ellipsis"
                      >
                        <div
                          class="ant-flex css-dev-only-do-not-override-1okl62o"
                        >
                          <div
                            class="css-x3zji3 ant-flex css-dev-only-do-not-override-1okl62o"
                          >
                            <span
                              class="ant-avatar ant-avatar-circle ant-avatar-image css-dev-only-do-not-override-1okl62o"
                            >
                              <img
                                src="https://randomuser.me/api/portraits/thumb/men/80.jpg"
                              />
                            </span>
                          </div>
                          <div>
                            <div
                              class="css-g90fw7"
                            >
                              Mr
                               
                              Sebastian
                               
                              Silva
                            </div>
                            <span
                              class="css-q3ox3x"
                            >
                              sebastian.silva@example.com
                            </span>
                          </div>
                        </div>
                      </td>
                      <td
                        class="ant-table-cell ant-table-cell-ellipsis"
                      >
                        <div
                          class="css-9ur1b ant-flex css-dev-only-do-not-override-1okl62o"
                        >
                          <div
                            class="css-x3zji3 ant-flex css-dev-only-do-not-override-1okl62o"
                          >
                            <span
                              aria-label="environment"
                              class="anticon anticon-environment"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="environment"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M854.6 289.1a362.49 362.49 0 00-79.9-115.7 370.83 370.83 0 00-118.2-77.8C610.7 76.6 562.1 67 512 67c-50.1 0-98.7 9.6-144.5 28.5-44.3 18.3-84 44.5-118.2 77.8A363.6 363.6 0 00169.4 289c-19.5 45-29.4 92.8-29.4 142 0 70.6 16.9 140.9 50.1 208.7 26.7 54.5 64 107.6 111 158.1 80.3 86.2 164.5 138.9 188.4 153a43.9 43.9 0 0022.4 6.1c7.8 0 15.5-2 22.4-6.1 23.9-14.1 108.1-66.8 188.4-153 47-50.4 84.3-103.6 111-158.1C867.1 572 884 501.8 884 431.1c0-49.2-9.9-97-29.4-142zM512 880.2c-65.9-41.9-300-207.8-300-449.1 0-77.9 31.1-151.1 87.6-206.3C356.3 169.5 431.7 139 512 139s155.7 30.5 212.4 85.9C780.9 280 812 353.2 812 431.1c0 241.3-234.1 407.2-300 449.1zm0-617.2c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm79.2 255.2A111.6 111.6 0 01512 551c-29.9 0-58-11.7-79.2-32.8A111.6 111.6 0 01400 439c0-29.9 11.7-58 32.8-79.2C454 338.6 482.1 327 512 327c29.9 0 58 11.6 79.2 32.8C612.4 381 624 409.1 624 439c0 29.9-11.6 58-32.8 79.2z"
                                />
                              </svg>
                            </span>
                          </div>
                          <div>
                            Boise
                            , 
                            <span
                              class="css-q3ox3x"
                            >
                              New Jersey
                            </span>
                          </div>
                        </div>
                      </td>
                      <td
                        class="ant-table-cell ant-table-cell-ellipsis"
                      >
                        <div
                          class="css-9ur1b ant-flex css-dev-only-do-not-override-1okl62o"
                        >
                          <div
                            class="css-x3zji3 ant-flex css-dev-only-do-not-override-1okl62o"
                          >
                            <span
                              aria-label="calendar"
                              class="anticon anticon-calendar"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="calendar"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                                />
                              </svg>
                            </span>
                          </div>
                          <div>
                            <span
                              style="margin-right: 8px;"
                            >
                              05/21/1956
                            </span>
                            <span
                              class="css-q3ox3x"
                            >
                              (Age 
                              67
                              )
                            </span>
                          </div>
                        </div>
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        <div
                          class="css-9ur1b ant-flex css-dev-only-do-not-override-1okl62o"
                          style="align-items: center;"
                        >
                          <div
                            class="css-x3zji3 ant-flex css-dev-only-do-not-override-1okl62o"
                          >
                            <span
                              aria-label="phone"
                              class="anticon anticon-phone"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="phone"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M877.1 238.7L770.6 132.3c-13-13-30.4-20.3-48.8-20.3s-35.8 7.2-48.8 20.3L558.3 246.8c-13 13-20.3 30.5-20.3 48.9 0 18.5 7.2 35.8 20.3 48.9l89.6 89.7a405.46 405.46 0 01-86.4 127.3c-36.7 36.9-79.6 66-127.2 86.6l-89.6-89.7c-13-13-30.4-20.3-48.8-20.3a68.2 68.2 0 00-48.8 20.3L132.3 673c-13 13-20.3 30.5-20.3 48.9 0 18.5 7.2 35.8 20.3 48.9l106.4 106.4c22.2 22.2 52.8 34.9 84.2 34.9 6.5 0 12.8-.5 19.2-1.6 132.4-21.8 263.8-92.3 369.9-198.3C818 606 888.4 474.6 910.4 342.1c6.3-37.6-6.3-76.3-33.3-103.4zm-37.6 91.5c-19.5 117.9-82.9 235.5-178.4 331s-213 158.9-330.9 178.4c-14.8 2.5-30-2.5-40.8-13.2L184.9 721.9 295.7 611l119.8 120 .9.9 21.6-8a481.29 481.29 0 00285.7-285.8l8-21.6-120.8-120.7 110.8-110.9 104.5 104.5c10.8 10.8 15.8 26 13.3 40.8z"
                                />
                              </svg>
                            </span>
                          </div>
                          <div>
                            <div>
                              (848) 863-4315
                               
                              <span
                                class="css-q3ox3x"
                              >
                                (m)
                              </span>
                               
                            </div>
                            <div>
                              (329) 449-6469
                               
                              <span
                                class="css-q3ox3x"
                              >
                                (p)
                              </span>
                            </div>
                          </div>
                        </div>
                      </td>
                      <td
                        class="ant-table-cell ant-table-cell-ellipsis"
                      >
                        <div
                          class="css-9ur1b ant-flex css-dev-only-do-not-override-1okl62o"
                        >
                          <div
                            class="css-x3zji3 ant-flex css-dev-only-do-not-override-1okl62o"
                          >
                            <span
                              aria-label="calendar"
                              class="anticon anticon-calendar"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="calendar"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                                />
                              </svg>
                            </span>
                          </div>
                          <div>
                            <span
                              style="margin-right: 8px;"
                            >
                              06/15/2021
                            </span>
                            <span
                              class="css-q3ox3x"
                            >
                              (
                              2
                               years ago)
                            </span>
                          </div>
                        </div>
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
          <ul
            class="ant-pagination ant-table-pagination ant-table-pagination-right css-dev-only-do-not-override-1okl62o"
          >
            <li
              aria-disabled="true"
              class="ant-pagination-prev ant-pagination-disabled"
              title="Previous Page"
            >
              <button
                class="ant-pagination-item-link"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <span
                  aria-label="left"
                  class="anticon anticon-left"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="left"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                    />
                  </svg>
                </span>
              </button>
            </li>
            <li
              class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
              tabindex="0"
              title="1"
            >
              <a
                rel="nofollow"
              >
                1
              </a>
            </li>
            <li
              class="ant-pagination-item ant-pagination-item-2"
              tabindex="0"
              title="2"
            >
              <a
                rel="nofollow"
              >
                2
              </a>
            </li>
            <li
              class="ant-pagination-item ant-pagination-item-3"
              tabindex="0"
              title="3"
            >
              <a
                rel="nofollow"
              >
                3
              </a>
            </li>
            <li
              class="ant-pagination-item ant-pagination-item-4"
              tabindex="0"
              title="4"
            >
              <a
                rel="nofollow"
              >
                4
              </a>
            </li>
            <li
              class="ant-pagination-item ant-pagination-item-5 ant-pagination-item-before-jump-next"
              tabindex="0"
              title="5"
            >
              <a
                rel="nofollow"
              >
                5
              </a>
            </li>
            <li
              class="ant-pagination-jump-next ant-pagination-jump-next-custom-icon"
              tabindex="0"
              title="Next 5 Pages"
            >
              <a
                class="ant-pagination-item-link"
              >
                <div
                  class="ant-pagination-item-container"
                >
                  <span
                    aria-label="double-right"
                    class="anticon anticon-double-right ant-pagination-item-link-icon"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="double-right"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M533.2 492.3L277.9 166.1c-3-3.9-7.7-6.1-12.6-6.1H188c-6.7 0-10.4 7.7-6.3 12.9L447.1 512 181.7 851.1A7.98 7.98 0 00188 864h77.3c4.9 0 9.6-2.3 12.6-6.1l255.3-326.1c9.1-11.7 9.1-27.9 0-39.5zm304 0L581.9 166.1c-3-3.9-7.7-6.1-12.6-6.1H492c-6.7 0-10.4 7.7-6.3 12.9L751.1 512 485.7 851.1A7.98 7.98 0 00492 864h77.3c4.9 0 9.6-2.3 12.6-6.1l255.3-326.1c9.1-11.7 9.1-27.9 0-39.5z"
                      />
                    </svg>
                  </span>
                  <span
                    class="ant-pagination-item-ellipsis"
                  >
                    •••
                  </span>
                </div>
              </a>
            </li>
            <li
              class="ant-pagination-item ant-pagination-item-100"
              tabindex="0"
              title="100"
            >
              <a
                rel="nofollow"
              >
                100
              </a>
            </li>
            <li
              aria-disabled="false"
              class="ant-pagination-next"
              tabindex="0"
              title="Next Page"
            >
              <button
                class="ant-pagination-item-link"
                tabindex="-1"
                type="button"
              >
                <span
                  aria-label="right"
                  class="anticon anticon-right"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="right"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                    />
                  </svg>
                </span>
              </button>
            </li>
            <li
              class="ant-pagination-options"
            >
              <div
                aria-label="Page Size"
                class="ant-select ant-select-outlined ant-pagination-options-size-changer css-dev-only-do-not-override-1okl62o ant-select-single ant-select-show-arrow ant-select-show-search"
              >
                <div
                  class="ant-select-selector"
                >
                  <span
                    class="ant-select-selection-search"
                  >
                    <input
                      aria-autocomplete="list"
                      aria-controls="rc_select_TEST_OR_SSR_list"
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      aria-label="Page Size"
                      aria-owns="rc_select_TEST_OR_SSR_list"
                      autocomplete="off"
                      class="ant-select-selection-search-input"
                      id="rc_select_TEST_OR_SSR"
                      role="combobox"
                      type="search"
                      value=""
                    />
                  </span>
                  <span
                    class="ant-select-selection-item"
                    title="10 / page"
                  >
                    10 / page
                  </span>
                </div>
                <span
                  aria-hidden="true"
                  class="ant-select-arrow"
                  style="user-select: none;"
                  unselectable="on"
                >
                  <span
                    aria-label="down"
                    class="anticon anticon-down ant-select-suffix"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </main>
</div>
`;
